presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -338,6 +338,12 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
                     updateMode = UpdateMode.NEW;
                     writeInfo = locationService.getPartitionWriteInfo(locationHandle, partition, partitionName.get());
                 }
+                else if (table.isTemporary()) {
+                    // this assumes that temporary table is always empty at this step
+                    // multiple inserts into temporary table are not allowed on the metadata level
+                    updateMode = UpdateMode.APPEND;
```

Comment:
If inserting into unpartitioned table is not allowed, I guess this insertion attempt would fail?

Reply:
I had to remove that restriction explicitly for temporary tables